ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp

```diff
@@ -1789,17 +1789,25 @@ Tensor change_layout_to_tile(const Tensor& temp, const MemoryConfig& output_mem_
 std::vector<Tensor> ExecuteUnaryBackwardProd::invoke(
     const Tensor& grad,
     const Tensor& input,
-    bool all_dimensions,
-    int64_t dim,
+    const std::optional<int64_t> dim,
     const std::optional<MemoryConfig>& output_mem_config) {
     std::vector<Tensor> grad_tensor;
     auto output_memory_config = output_mem_config.value_or(
         input.memory_config());  // TODO: Remove after ternary forward ops migration is completed
-    const bool keepdim = !all_dimensions;
-    Tensor prod_result = ttnn::prod(input, all_dimensions, dim, keepdim, output_memory_config);
+
+    Tensor prod_result;
+    const bool all_dimensions = !dim.has_value();
+
+    if (all_dimensions) {
+        prod_result = ttnn::prod(input, /*dim=*/std::nullopt, /*keepdim=*/false, output_memory_config);
+    } else {
+        prod_result = ttnn::prod(input, *dim, /*keepdim=*/true, output_memory_config);
+    }
```

Comment:
I like that you've combined the two parameters into an optional. I'm trying to understand this change though, is this meant to be equivalent behavior, or is anything intentionally changed here?